docs/deep-dive/cli/your-first-cli-plugin.md

```diff
@@ -0,0 +1,54 @@
+---
+id: your-first-cli-plugin
+title: Your first CLI plugin
+sidebar_label: Your first CLI plugin
+---
+
+This guide will help you create your own CLI plugins using our [Plugins system](/docs/deep-dive/plugins-crash-course).
+
+This process is very similar to [creating scaffold plugins](/docs/deep-dive/scaffolding/your-first-scaffold-plugin).
+
+We use [yargs](https://www.npmjs.com/package/yargs) to create our CLI, so you should be somewhat familiar with it.
+
+## Here's how to do it
```

Comment:
This title should be changed to something more formal
maybe:
"Where to start"

Reply:
done